### PR TITLE
Migrate to Bzlmod

### PR DESCRIPTION
**Pull request by jadenPete**

### CODE REVIEW

File: MODULE.bazel
Comment (jadenPete) [inline MODULE.bazel]: - [ ] Wait until we've merged https://github.com/lucidsoftware/rules_scala/pull/74 and update this.

File: .bazelrc
Comment (jjudd) [inline .bazelrc:3]: Is this something all users of Annex will need to do? If so, is it in the README for that ruleset?

File: MODULE.bazel
Comment (jjudd) [inline MODULE.bazel]: Can we mark all of the deps that we can as `dev_dependency`? I think this is one.

File: MODULE.bazel
Comment (jjudd) [inline MODULE.bazel]: I think this is another `dev_dependency`

File: MODULE.bazel
Comment (jjudd) [inline MODULE.bazel]: Is this dep intentionally below the others or can it be moved up with the others?

Also I think this can be marked as `dev_dependency`

File: MODULE.bazel
Comment (jjudd) [inline MODULE.bazel]: Scope creep, but could we update this to 1.10.4 while we're at it? Annex is already on that version, so it shouldn't require anymore work besides just updating the number and generating the lock file.

**Review by jjudd — COMMENTED** [review]
Mostly LGTM. Left a few comments.

File: .bazelrc
Comment (jadenPete) [reply .bazelrc:3]: Nope, this is only necessary if you want to override the default Scala toolchain. It's explained here:
https://github.com/lucidsoftware/rules_scala/blob/74960c12dfc2487cd257c2a49d5d4e1c6cfb1d44/docs/newdocs/scala_versions.md

Let me know if I can make it clearer in any way.

File: MODULE.bazel
Comment (jadenPete) [reply MODULE.bazel]: See my comment here:
https://github.com/lucidsoftware/rules_scala/pull/74#discussion_r1873510258

File: MODULE.bazel
Comment (jadenPete) [reply MODULE.bazel]: See my comment here:
https://github.com/lucidsoftware/rules_scala/pull/74#discussion_r1873510258

File: MODULE.bazel
Comment (jadenPete) [reply MODULE.bazel]: I wanted to keep the `archive_override` call close to the `bazel_dep` call for `rules_scala_annex`. That way, it's not confusing where `rules_scala_annex` is coming from or what version we're using.

Also, see my comment here:
https://github.com/lucidsoftware/rules_scala/pull/74#discussion_r1873510258

File: MODULE.bazel
Comment (jadenPete) [reply MODULE.bazel]: Sure thing. I refrained from updating any versions in the `lucidsoftware/rules_scala_annex` PR because I considered it scope creep, but this is a simple change.

File: MODULE.bazel
Comment (jjudd) [reply MODULE.bazel]: Ah got it. Makes sense.

File: MODULE.bazel
Comment (jjudd) [reply MODULE.bazel]: Thanks

**Review by jjudd — APPROVED** [review]

File: MODULE.bazel
Comment (jadenPete) [reply MODULE.bazel]: No problem!

**Comment by jadenPete** [comment]
Squashed.

**Pull request merged**
